components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/Utils.java

```diff
@@ -63,10 +63,13 @@ public static void sendFault(MessageContext messageContext, int status) {
         messageContext.setTo(null);        
         axis2MC.removeProperty("NO_ENTITY_BODY");
         String method = (String) axis2MC.getProperty(Constants.Configuration.HTTP_METHOD);
-        if (method.matches("^(?!.*(POST|PUT|PATCH)).*$")) {
+
+        /* commented due to, fault message type need to be request accept type */
+        /*if (method.matches("^(?!.*(POST|PUT|PATCH)).*$")) {
             // If the request was not an entity enclosing request, send a XML response back
             axis2MC.setProperty(Constants.Configuration.MESSAGE_TYPE, "application/xml");
-        }
+        }*/
```

Comment:
fixed in commit: https://github.com/wso2/carbon-apimgt/commit/9619da2271b49f37ef93456b6f8a8ac235f685ea